storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -102,49 +102,14 @@ public String topicWarningMessage(String topicName) {
 
     public static class RemoteLogConfig {
```

Comment:
could we please add an assertion and a test that remote storage cannot be enabled for a topic if the remote storage is not enabled at system level? Please let me know if I am missing something and we already have this validation check somewhere else.

Reply:
We cannot infer KafkaConfig from LogConfig class.

Filed KAFKA-15290 to initialise/stop the remote log components for a topic when `remote.storage.enable` is set to true/false. In [TopicConfigHandler](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ConfigHandler.scala#L55C13-L55C13), we can add this validation but the config might already saved in the ConfigRepository.

Do you have suggestion on how to add this validation?

Reply:
Thanks for the suggestion! Can we take this validation separately? If yes, will file a ticket and work on it.

This patch is about adding broker-level config synonyms. Including this validation in the PR will increase the scope and delay the PR from landing it. WDYT?

Reply:
sure. separate ticket works.

Reply:
Filed KAFKA-15295 to add this validation. Thanks for reviewing the PR!

Reply:
Opened #14176 to address this comment. PTAL.

Reply:
I think we didn't fix the similar issue in [TopicCommand.java#L173](https://github.com/apache/kafka/blame/3.7/tools/src/main/java/org/apache/kafka/tools/TopicCommand.java#L173). In `LogConfig.validate(props);` call `isRemoteLogStorageSystemEnabled` is always false. 

Please let me know if I am wrong here